compiler-rt/test/ctx_profile/TestCases/generate-context.cpp

```diff
@@ -46,22 +44,31 @@ __attribute__((noinline)) void theRoot() {
 // CHECK-NEXT: check even
 // CHECK-NEXT: check odd
 
-void printProfile(const ContextNode &Node, const std::string &Indent,
-                  const std::string &Increment) {
-  std::cout << Indent << "Guid: " << Node.guid() << std::endl;
-  std::cout << Indent << "Entries: " << Node.entrycount() << std::endl;
-  std::cout << Indent << Node.counters_size() << " counters and "
-            << Node.callsites_size() << " callsites" << std::endl;
-  std::cout << Indent << "Counter values: ";
-  for (uint32_t I = 0U; I < Node.counters_size(); ++I)
-    std::cout << Node.counters()[I] << " ";
-  std::cout << std::endl;
-  for (uint32_t I = 0U; I < Node.callsites_size(); ++I)
-    for (const auto *N = Node.subContexts()[I]; N; N = N->next()) {
-      std::cout << Indent << "At Index " << I << ":" << std::endl;
-      printProfile(*N, Indent + Increment, Increment);
-    }
-}
+class TestProfileWriter : public ProfileWriter {
+  void printProfile(const ContextNode &Node, const std::string &Indent,
```

Comment:
I think it might be usefult to pass in a `llvm::raw_ostream&` and then writing to it. I think this will give you more flexibility based when used in tests (e.g using a string buffer).

Reply:
This is a .cpp file that gets compiled (and run) as part of the test, so we basically have access to `std::` things.

Reply:
Just to make sure we're talking about the same thing: this here is a lit test. Note the RUN lines at the top of the file.

Reply:
Discussed offline and since this in compiler-rt its not useful to use ostream.